src/test/java/gov/healthit/chpl/aqa/asserts/ListingHistoryEyeAsserts.java

```diff
@@ -0,0 +1,64 @@
+package gov.healthit.chpl.aqa.asserts;
+
+import static org.junit.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+import java.util.Calendar;
+import java.util.Date;
+
+import cucumber.api.java.en.Then;
+import gov.healthit.chpl.aqa.model.ListingHistoryData;
+import gov.healthit.chpl.aqa.pageObjects.ListingHistoryEyePage;
+import gov.healthit.chpl.aqa.stepDefinitions.Base;
+
+/**
+ * Class ListingHistoryEyeAsserts definition.
+ */
+public class ListingHistoryEyeAsserts extends Base {
+
+    /**
+     * Assert that expected product edit details is displayed on product history eye.
+     * @param expectedEditDetails is "Developer/Product/Version change"
+     * @throws Exception of type ParseException or InterruptedException
+     */
+    @Then("^I see the listing history eye shows a \"([^\"]*)\" on today's date$")
+    public void iSeeListingHistory(final String expectedEditDetails) throws Exception {
+        Thread.sleep(SLEEP_TIME);
+        String actualListingHistoryDetails = ListingHistoryEyePage.listingHistoryTable(getDriver()).getText();
+        String[] listingHistoryDetails = actualListingHistoryDetails.split("\n");
+        int activityCount = 0;
+        ListingHistoryData listingHistoryDataInst = null;
+        String date = null;
+        for (String activity : listingHistoryDetails) {
+            if ((activityCount % 2) == 0) {
+                date = activity;
+            } else {
+                if (activity.startsWith(expectedEditDetails)) {
+                    listingHistoryDataInst = new ListingHistoryData();
+                    listingHistoryDataInst.setDate(date);
+                    listingHistoryDataInst.setActivity(activity);
+                    break;
+                }
+            }
+            activityCount++;
+        }
+        if (listingHistoryDataInst == null) {
+            assertFalse("No edit activity recorded for today", true);
+        }
+        Calendar calInstanceExpectedDate = Calendar.getInstance();
+        Calendar calInstanceCurrentDate = Calendar.getInstance();
+        Date actualDate = getFormatedDateTimeZone(listingHistoryDataInst.getDate());
+        Date currentDate = getFormatedDateTimeZone(null);
+        calInstanceExpectedDate.setTime(actualDate);
+        calInstanceCurrentDate.setTime(currentDate);
+        boolean listingHistoryRecorded = false;
+        if ((calInstanceExpectedDate.get(Calendar.YEAR) == calInstanceCurrentDate.get(Calendar.YEAR))
```

Comment:
Don't use the calendarInstance to compare things

```
Date foundDate = getFormatedThing(string from the table)
Date actualDate = new Date
if (actualDate.getTime - foundDate.getTime < some number of milliseconds) {
// then the "current time" is less than "some number of milliseconds" after the "found date stamp", and your activity happened "recently"
```

You're going from a string, to a date, to a calendar instance, and then only comparing calendars down to the day of month resolution, which means that if you ran this test "now", and it worked, and then you ran it again in a few minutes, and something broke in the update code in the meantime, your test would still pass, even though the update might not have worked